From 5aef87b6c7ec51cefcc5b4f7633165ff5abe28b1 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 8 Jan 2014 15:06:14 +0800
Subject: [PATCH 33/37] ENGR00294124 Multi-user:System reboot when setup a new
 profile if the language is Chinese

Add a check on default IME, in case there is no default IME for some language setting
The exception is as below:
Found no subtypes in a system IME: com.android.inputmethod.pinyin
*** FATAL EXCEPTION IN SYSTEM PROCESS: ActivityManager
java.lang.NullPointerException
at com.android.server.InputMethodManagerService.resetDefaultImeLocked(InputMethodManagerService.java:736)
at com.android.server.InputMethodManagerService.resetAllInternalStateLocked(InputMethodManagerService.java:768)
at com.android.server.InputMethodManagerService.switchUserLocked(InputMethodManagerService.java:805)
at com.android.server.InputMethodManagerService.access$1000(InputMethodManagerService.java:127)
at com.android.server.InputMethodManagerService$3.onUserSwitching(InputMethodManagerService.java:656)

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 services/java/com/android/server/InputMethodManagerService.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/java/com/android/server/InputMethodManagerService.java b/services/java/com/android/server/InputMethodManagerService.java
index a996dbd..a94b1bd 100644
--- a/services/java/com/android/server/InputMethodManagerService.java
+++ b/services/java/com/android/server/InputMethodManagerService.java
@@ -733,7 +733,8 @@ public class InputMethodManagerService extends IInputMethodManager.Stub
         if (defIm == null && mMethodList.size() > 0) {
             defIm = InputMethodUtils.getMostApplicableDefaultIME(
                     mSettings.getEnabledInputMethodListLocked());
-            Slog.i(TAG, "No default found, using " + defIm.getId());
+            if (defIm != null)
+                Slog.i(TAG, "No default found, using " + defIm.getId());
         }
         if (defIm != null) {
             setSelectedInputMethodAndSubtypeLocked(defIm, NOT_A_SUBTYPE_ID, false);
-- 
1.8.0

